datafusion/physical-expr/src/expressions/in_list.rs

```diff
@@ -504,6 +634,11 @@ impl PhysicalExpr for InListExpr {
                         .unwrap();
                     set_contains_with_negated!(array, set, self.negated)
                 }
+                DataType::Decimal(_, _) => {
+                    let array = array.as_any().downcast_ref::<DecimalArray>().unwrap();
```

Comment:
Stylistic suggestion: you can also use https://docs.rs/arrow/16.0.0/arrow/array/fn.as_decimal_array.html here too

```suggestion
                    let array = as_decimal_array(array);
```

Reply:
got it.
But i will keep the original style